stdlib/public/core/Random.swift

```diff
@@ -94,15 +94,22 @@ extension RandomNumberGenerator {
     upperBound: T
   ) -> T {
     _precondition(upperBound != 0, "upperBound cannot be zero.")
-    let tmp = (T.max % upperBound) + 1
-    let range = tmp == upperBound ? 0 : tmp
-    var random: T = 0
-
-    repeat {
-      random = next()
-    } while random < range
-
-    return random % upperBound
+    var random: T = next()
+    var m = random.multipliedFullWidth(by: upperBound)
+    if (m.low < upperBound) {
```

Comment:
Nit:

```suggestion
    if m.low < upperBound {
```

...and elsewhere (to remove some C-isms).